package/MDAnalysis/auxiliary/base.py

```diff
@@ -482,6 +484,21 @@ def attach_auxiliary(self,
             coord_parent._auxs[auxname] = aux
             coord_parent.ts = aux.update_ts(coord_parent.ts)
 
+        aux_memory_usage = 0
+        # Check if testing, which needs lower memory limit
+        if "memory_limit" in kwargs:
+            memory_limit = kwargs["memory_limit"]
+        else:
+            memory_limit = 1e+09
+        for reader in coord_parent._auxs.values():
+            aux_memory_usage += reader._memory_usage()
+        if aux_memory_usage > memory_limit:
+            warnings.warn("AuxReader: memory usage warning!")
+
+    def _memory_usage(self):
```

Comment:
Its at times like this that Python needs `virtual` and `override` like in C++, but I like the design. :)